### ISSUE TITLE
Disconnect notion of failure from future

### ISSUE THREAD

**Issue by TheLudd** [ramda/ramda-fantasy#73]
Straight out of https://github.com/ramda/ramda-fantasy/pull/70#issuecomment-140719135 but contiuing the discussion here:

I find this very interesting. It has bothered me that there are two ways of handling errors, Future and Either. What would be gained from removing the forking from futures and replacing it with `run` or something similar?

**Comment by TheLudd** [comment]
If this is to be done then we *must* make a Future transformer. Is that correct?  Instead of the current `Future` one would use `Future.T(Either)` right?

But is it possible? Consider the `chain` method of `Future.T`. I have made a cheating implementation here: https://github.com/TheLudd/ramda-fantasy/blob/future-transformer/src/Future.js#L139
```
  FutureT.prototype.chain = function(f) {
    var futureT = this;
    return new FutureT(function(reject, resolve) {
      return futureT.fork(reject, function(v) {
        var v2 = v.chain(f);
        if(v2 instanceof M) {
          resolve(v2);
        } else {
          v2.fork(reject, resolve);
        }
      });
    });
  };
```
The problem is that `f` returns type `Future.T(M)`, but `v.chain` expects a function that returns type `M`. There is nothing in the FL spec that says that `f` must be called so `Nothing` and `Left` will return an instance of their own type. `Just` and others will return whatever `f` returns. 

My implementation works for `Future.T(Maybe)`, `Future.T(Either)` but it is cheating since the call `v.chain(f)` breaks the contract of `chain` and I think this will cause problems. 

Is there any way of solving this without cheating? Would things change if we did not fork but simply had ` run(cb)`? I don't believe it would. 

I would however love a Future transformer..
